poetro_app/lib/features/poetry/data/data_sources/remote/poetry_remote_data_source_impl.dart

```diff
@@ -41,10 +44,32 @@ class PoetryDataSource implements IPoetryRemoteDataSource {
   }
 
   @override
-  Future<List<PoetryDTO>> getPoetryListByKeyword(String keyword) async {
-    final response = await _dio.get(ApiEndpoints.poemByKeyword(keyword));
-
-    return (response.data as List).map((e) => PoetryDTO.fromJson(e)).toList();
+  Future<List<PoetryDTO>> getPoetryListByKeyword(
+      String keyword, int count) async {
+    try {
+      final response = await _dio.get(
+        ApiEndpoints.poemByKeyword(
+          keyword,
+          count,
+        ),
+      );
+
+      final list = response.data as List<dynamic>;
+
+      log(list.length.toString());
+
+      if (list.isEmpty) {
+        return [];
+      }
+
+      return list
+          .map(
+            (e) => PoetryDTO.fromJson(e as Map<String, dynamic>),
+          )
+          .toList();
+    } catch (e) {
+      throw ApiException(message: e.toString());
+    }
```

Comment:
error handling is bad here, you can use eaither and Error class for it ,
throw ApiException(message: e.toString()); this is some us by default if you will not do anything